pkg/controllers/allocation/controller.go

```diff
@@ -92,13 +101,15 @@ func (c *Controller) Reconcile(ctx context.Context, object client.Object) (recon
 		packings = append(packings, c.packer.Pack(ctx, constraintGroup, instanceTypes)...)
 	}
 
-	// 4. Create packedNodes for packings
-	packedNodes, err := c.cloudProvider.Create(ctx, provisioner, packings)
+	// 5. Create packedNodes for packings and also copy all Status changes made by the
+	//    cloud provider to the original provisioner instance.
+	packedNodes, err := c.cloudProvider.Create(ctx, &provisionerWithDefaults, packings)
+	provisioner.Status = provisionerWithDefaults.Status
```

Comment:
I think we should skip this functionality for now. We started off with some status conditions which were carried forward from some legacy code in Karpenter's original POC. I want to revisit status and conditions more intentionally in the future.